sphinxcontrib/spelling/filters.py

```diff
@@ -190,14 +190,15 @@ def _skip(self, word):
         if word not in self.sought_modules:
             self.sought_modules.add(word)
             try:
-                imp.find_module(word)
-            except UnicodeEncodeError:
```

Comment:
Do we no longer expect to see this exception type because of the change to importlib? Or Python 3?

Reply:
I'm not sure which, but AFAICT, the need to handle this exception is obsolete.

`importlib` on Python 3 handles both files with invalid encoding as well it handles filename strings with invalid . Here is my testing:

File with invalid utf-8:

```sh
$ echo -e '\x81' > broken.py
$ python3 broken.py
  File "broken.py", line 1
SyntaxError: Non-UTF-8 code starting with '\x81' in file broken.py on line 1, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details

# Above shows the file is invalid Python.

$ python3 -c 'import importlib; spec = importlib.util.find_spec("broken"); print(spec)'
ModuleSpec(name='broken', loader=<_frozen_importlib_external.SourceFileLoader object at 0x7f284f8313a0>, origin='/home/jon/test2/broken.py')

# Above shows that calling find_spec() on the same file doesn't raise an exception.
```

Filename with invalid encoding:


```sh
$ python3 -c '"\u0411".encode("iso-8859-15")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib64/python3.8/encodings/iso8859_15.py", line 12, in encode
    return codecs.charmap_encode(input,errors,encoding_table)
UnicodeEncodeError: 'charmap' codec can't encode character '\u0411' in position 0: character maps to <undefined>

# Above shows string can't be encoded.

$ python3 -c 'import sys, importlib; sys.getfilesystemencoding = lambda: "iso-8859-15"; spec = importlib.util.find_spec("\u0411"); print(spec)'

# Above shows the invalid string doesn't raise an exception.
```

Reply:
Nice analysis, thanks for working all of that out. I agree, it seems safe to drop that exception.